tests/testthat/test-nm-join.R

```diff
@@ -165,3 +165,33 @@ test_that("nm_join() warns on skipping table with wrong number of rows [BBR-NMJ-
   expect_equal(nrow(test_df), DATA_TEST_ROWS)
   expect_equal(ncol(test_df), DATA_TEST_COLS)
 })
+
+test_that("Confirming unduplicates rows on .join_col [BBR-NMJ-007]",{
+  withr::with_options(list(bbr.bbi_exe_path = read_bbi_path()), {
+  withr::with_tempdir({
+    fs::dir_copy(system.file("model","nonmem", "basic",package = "bbr"), file.path(tempdir(),"basic"))
+    on.exit(if( fs::dir_exists(file.path(tempdir(), "basic"))) fs::dir_delete(file.path(tempdir(), "basic")))
+    fs::file_copy(system.file("extdata", "acop.csv", package = "bbr"), file.path(tempdir(), "basic", "1"), overwrite = TRUE)
+
+
+    #Edit Table File to Create Duplicate NUM Row
+    file.path(tempdir(), "basic", "1", "1.tab") %>% read_lines() %>%
+      str_replace("2.0000E","1.0000E") %>% write_lines(file.path(tempdir(), "basic", "1", "1.tab"))
+
+    json <- jsonlite::read_json(file.path(tempdir(), "basic", "1", "bbi_config.json"))
+
+    json$data_path <- "acop.csv"
+
+    jsonlite::write_json(json,file.path(tempdir(), "basic", "1", "bbi_config.json"))
+
+    expect_error(file.path(tempdir(), "basic", "1") %>% nm_join(.files = "1.tab"))
+
+    fs::file_copy(system.file("model","nonmem", "basic","1", "1.tab",package = "bbr"),
+                  file.path(tempdir(), "basic", "1", "1.tab"), overwrite = TRUE)
+    .d <- readr::read_csv(file.path(tempdir(), "basic", "1", "acop.csv"))
+    .d$num[2] <- 1
+    readr::write_csv(.d, file.path(tempdir(), "basic", "1", "acop.csv"))
+
+    expect_error(file.path(tempdir(), "basic", "1") %>% nm_join(.files = "1.tab"))
```

Comment:
In these expect_error calls, can you specify the first few words/a phrase of what error is expected

Reply:
When I run the first expect_error case (without the expectation), I get:
```r
> file.path(tempdir(), "basic", "1") %>% nm_join(.files = "1.tab")
Reading data file: acop.csv
  rows: 799
  cols: 10

Reading 1.tab
  rows: 779
  cols: 8

tab adds 7 new cols                                                                                                                                                                                  

Duplicate rows in NUM: c(1, 100)
 Error in drop_dups(tab, .d, .join_col, .n) : 
```

instead of `"Duplicate rows in data in {.join_col}: {dup_row}"`

i.e. Its printing out your message, but not as an error:

```r
>     expect_error(file.path(tempdir(), "basic", "1") %>% nm_join(.files = "1.tab"), "Duplicate rows in data")
Reading data file: acop.csv
  rows: 799
  cols: 10

Reading 1.tab
  rows: 779
  cols: 8

Duplicate rows in data in NUM: 1
 Error: `file.path(tempdir(), "basic", "1") %>% nm_join(.files = "1.tab")` threw an error with unexpected message.
Expected match: "Duplicate rows in data"
Actual message: "" 
```

However, when I get rid of `verbose_message()`, we get the expected message:
```r
> file.path(tempdir(), "basic", "1") %>% nm_join(.files = "1.tab")
Reading data file: acop.csv
  rows: 799
  cols: 10

Reading 1.tab
  rows: 779
  cols: 8

 Error in nm_join(., .files = "1.tab") : Duplicate rows in data in NUM: 1 
```

```r
>     expect_error(file.path(tempdir(), "basic", "1") %>% nm_join(.files = "1.tab"), "Duplicate rows in data")
Reading data file: acop.csv
  rows: 799
  cols: 10

Reading 1.tab
  rows: 779
  cols: 8
```

Did anyone tell you to use `verbose_message()` instead of glue?